bot/engine/src/main/kotlin/engine/config/tickstory/TickAnswerHandler.kt

```diff
@@ -0,0 +1,127 @@
+
+            // Get a stored tick state. Start a new session if it doesn't exist
+            val tickSession = initTickSession(dialog, story._id.toString(), connectorData.conversationData)
+
+            // Call the tick story processor
+            val result =
+                TickStoryProcessor(
+                    tickSession,
+                    configuration.toTickConfiguration(),
+                    TickSenderBotBus(botBus),
+                    endingStoryRuleExists
+                ).process(
+                    TickUserAction(
+                        intentName,
+                        parseEntities(entities, tickSession.init))
+                )
+
+            when (result) {
+                is Success -> updateDialog(dialog, story._id.toString(), result.session)
+                is Redirect -> {
+                    // when a redirection is performed, the current story state is considered as finished
+                    dialog.tickStates.compute(story._id.toString()) { _, state -> state?.copy(finished = true) }
+
+                    result.storyId?.let { redirectFn(it) }
+                }
+            }
+
+        }
+    }
+
+    /**
+     * If action is final, then remove a given tick story from a Dialog
+     * else update a tick state
+     */
+    private fun updateDialog(dialog: Dialog, storyId: String, tickSession: TickSession) {
+        logger.debug { "updating dialog tick state with session data... " }
+        dialog.tickStates[storyId] =
+            with(tickSession) {
+                TickState(
+                    currentState!!,
+                    contexts,
+                    ranHandlers,
+                    objectivesStack,
+                    init,
+                    unknownHandlingStep,
+                    handlingStep,
+                    finished
+                )
+            }
+    }
+
+    /**
+     * Initialize a tick session
+     */
+    private fun initTickSession(dialog: Dialog, storyId: String, conversationData: Map<String, String>): TickSession {
+        val tickState = dialog.tickStates[storyId]
+        return with(tickState) {
+            if (this == null) {
+                TickSession(init = dialog.lastDateUpdate, contexts = conversationData)
+            } else if (finished) {
+                TickSession(init = dialog.lastDateUpdate, contexts = conversationData)
+            } else {
+                var sessionCtx = contexts
+                conversationData.forEach { (t, u) ->
+                    if (!contexts.containsKey(t))
+                        sessionCtx = contexts + (t to u)
+                }
+                TickSession(currentState, sessionCtx, ranHandlers, objectivesStack, init, unknownHandlingStep, handlingStep)
+            }
+        }
+    }
+    /**
+     * Feed the contexts by the entities provided from the initialization date
+     */
+    private fun parseEntities(entities: Map<String, EntityStateValue>, init: Instant): Map<String, String?> {
```

Comment:
```suggestion
    private fun parseEntities(entities: Map<String, EntityStateValue>, init: Instant): Map<String, String?> {
```
init pas top comme nom de date car `init`, privilégier updateDate plutôt ?

Reply:
c'est la date d'initialisation de la session... je la renomme en initDate / initializationDate ?